src/Mesh/Interpolation.jl

```diff
@@ -996,13 +1025,59 @@ function interpolate_cubed_sphere_CUDA!(offset::AbstractArray{T,1}, m1_r::Abstra
                 end
             end
             @inbounds v[off + i, st_no] = vout_jk[1,1] * fc
+
+
         end
         #-----------------------------------
     end
     #-----------------------------------
     return nothing
 end
 #--------------------------------------------------------
+function project_cubed_sphere_CUDA!(lat_grd::AbstractArray{FT,1}, long_grd::AbstractArray{FT,1}, 
+                                       lati::AbstractVector{UInt16}, longi::AbstractVector{UInt16}, 
+                                          v::AbstractArray{FT}) where {FT <: AbstractFloat}
+
+    ti     = threadIdx().x # thread ids
+    bi     = blockIdx().x  # block ids 
+    bs     = blockDim().x  # block dim
+    idx    = ti + (bi-1)*bs
+    np_tot = size(v,1)
+
+    _ρu, _ρv, _ρw = 2, 3, 4
+
+    # projecting velocity onto unit vectors in rad, lat and long directions
+    # assumed u, v and w are located in columns 2, 3 and 4
+    if idx ≤ np_tot
+#        vrad =  v[idx,_ρu] * CUDAnative.cosd(lat_grd[lati[idx]]) * CUDAnative.cosd(long_grd[longi[idx]]) + 
```

Comment:
@vchuravy I'm assuming that this breaks because CUDA doesn't have a `cosd` function?

Reply:
In this case, the code does not break down. With the commented lines enabled, it runs to completion but produces the wrong result.  We ran the CUDA cosd function on a sample argument earlier today. While we noticed a difference, it was too small to make a big impact. Still trying to identify what is going wrong.

Reply:
Okay, we can keep what you have for now.